test/jdk/java/nio/file/WatchService/LotsOfEntries.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2010, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 8330077
+ * @summary Tests WatchService behavior with more entries in a watched directory
+ *     than the default event limit
+ * @library ..
+ * @run main/othervm LotsOfEntries 600 fail
+ * @run main/othervm -Dsun.nio.fs.maxWatchEvents=700 LotsOfEntries 600 pass
+ */
+
+import java.nio.file.*;
+import static java.nio.file.StandardWatchEventKinds.*;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class LotsOfEntries {
```

Comment:
What platforms have you tested this on? I'm concerned that it may fail intermittently on some platforms/configurations.

Reply:
It passes on Linux and macOS, I am in the process of testing it on Windows.

Reply:
It also passes on Windows 11.